framework/tests/flowexpression/InnerClasses.java

```diff
@@ -0,0 +1,38 @@
+package flowexpression;
+
+import testlib.flowexpression.qual.FlowExp;
+
+public class InnerClasses {
+    public String outerInstanceField = "";
+    public static String outerStaticField = "";
+
+    static class InnerClass {
+        // :: error: (expression.unparsable.type.invalid)
+        @FlowExp("outerInstanceField") Object o = null;
+
+        @FlowExp("outerStaticField") Object o2 = null;
+    }
+
+    class NonStaticInnerClass {
+        @FlowExp("outerInstanceField") Object o = null;
+
+        @FlowExp("outerStaticField") Object o2 = null;
+    }
+
+    static class InnerClass2 {
+        public String outerInstanceField = "";
+
+        @FlowExp("outerInstanceField") Object o = null;
+    }
+
+    class TestUses {
+        void method(InnerClass innerClass, InnerClass2 innerClass2) {
+            @SuppressWarnings("assignment.type.incompatible") // Issue 152
```

Comment:
Because the LHS annotation isn't parsable, this should issue an assignment.type.incompatible error.